### PR TITLE
[Questionable] db: improve db.has memdb usage

### PR DESCRIPTION
**Pull request by holiman**
I'm not too certain about this fix. It is an attempt to make `db.Get` use the `Contains` method from memdb, instead of retrieving the actual value. 
However, what makes me a bit uncertain about the correctness is that I don't fully understand the existing code, which does a `Find`, and then checks the returned key against the key we asked for. 
I don't understand why it doesn't simply do an `exact` check, instead of fetching "greater or equal" and then compares on the outside. 

Basically, I suspect that I'm missing something, pointers would be appreciated.

### CODE REVIEW

**Comment by holiman** [comment]
Ok, now I understand a bit more about the internal key, which is appended to the end of the userkey as a sequence number. So when we check the presence of a key, we need to find the position ourselves at the 'latest' such key and return the actual one, we can't just check the exact key. Closing this
